index.html

```diff
@@ -811,6 +814,123 @@ <h3>
           </p>
         </section>
       </section>
+      <section>
+        <h3>
+          `id` member
+        </h3>
+        <p>
+          The [=manifest's=] <code><dfn data-export="" data-dfn-for=
+          "manifest" id="identity">id</dfn></code> member is a <a>string</a> that
+          represents the <dfn>identity</dfn> for the application. 
+          The [=identity=] takes the form of a URL, which is same origin as the
+          start URL.
+        </p>
+        <p>The [=identity=] is used by user agents to uniquely identify the
+        application universally. When the user agent sees a manifest with an
+        [=identity=] that does not correspond to an already-installed application,
+        it SHOULD treat that manifest as a description of a distinct
+        application, even if it is served from the same URL as that of another
+        application. When the user agent sees a manifest with an [=identity=]
+        matching that of an already-installed application, it SHOULD be used as
+        a signal that this manifest is a replacement for the already-installed
+        application's manifest, and not a distinct application, even if it is
+        served from a different URL than the one seen previously.
+        </p>
+        <p>
+        The [=identity=] is considered matching when the processed |manifest|["id"] [=url/equal|equals=].
+        </p>
+        <p class="note">
+          The [=identity=] can be used by a service that collects lists of web
+          applications to uniquely identify applications.
+        </p>
+        <p class="note">
+          The [=identity=] is processed like a URL but it doesn't point to a
+          resource that can be navigated to, so it's not required to be
+          [=URL/within scope=].
+        </p>
+        <p>
+          To <dfn>process the `id` member</dfn>, given [=object=]
+          |json:JSON|, [=ordered map=] |manifest:ordered map|:
+        </p>
+        <ol class="algorithm">
+          <li>Set |manifest|["id"] to |manifest|["start_url"].
+          </li>
+          <li>If the type of |json|["id"] is not [=string=], return.
+          </li>
+          <li>If |json|["id"] is the empty string, return.
+          </li>
+          <li>Let |base origin| be |manifest|["start_url"]'s [=url/origin=].</li> 
+          <li>Let |id:URL| be the result of [=URL Parser|parsing=] |json|["id"] with  |base origin| as the base URL.
+          </li>
+          <li>If |id| is failure, return.
+          </li>
+          <li>If |id| is not [=same origin=] as |manifest|["start_url"], return.
+          </li>
+          <li>Otherwise, set |manifest|["id"] to |id|.</li>
```

Comment:
```suggestion
          <li>Remove [=URL/fragment=] from |id|.</li>
          <li>Set |manifest|["id"] to |id|.</li>
```

Reply:
You are right. I'd still be inclined to remove it from the id tho, just for simplicity. We could do that in step 2, and then again here?

Reply:
Just to be clear, I'd want to retain the same identity if I updated:


```JSON
{
    "start_url": "foo.html#here",
}
```

To:

```JSON
{
    "start_url": "foo.html#there",
}
```

And not consider that an entirely new application identity.

Reply:
I think as a developer, if I'd want to change the start_url but retaining the same identity, I should specify the id field instead?
```
{
  "start_url": "foo.html#here",
}
```
To:
```
{
  "start_url": "foo.html#there",
  "id": "foo.html#here",
}
```

If we remove the fragment, apps that have fragment in start_url today will get a new id when browser ships this feature.

Reply:
Sorry, I’m still not following? No app would have an identity today, as we haven’t settled on this feature yet?

Reply:
Right, for browsers that use start_url for the identity, this will change app identity from start_url to start_url with fragment removed.
You mentioned you think dropping the fragment is better for simplicity, but it seems to me that following existing processing logic for start_url is simpler?

Reply:
Ok I chatted more with dmurph@ offline. It makes sense to drop fragment considering #here or #there both link to the same page so they should be considered identical.
I will make the update to drop the fragment then!

Reply:
Thanks @philloooo. Appreciate you gathering the additional input from @dmurph. I've updated the Gecko implementation to match. 

I know it took us a a while, but really happy with were we are ending up with this 🥰